ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java

```diff
@@ -85,91 +95,101 @@ public static UnsignedLong get_previous_total_balance(BeaconState state) {
 
   /**
    * Returns the union of validator index sets, where the sets are the attestation participants of
-   * attestations passed in TODO: the union part takes O(n^2) time, where n is the number of
-   * validators. OPTIMIZE
+   * attestations passed in.
    */
   public static List<Integer> get_attesting_indices(
       BeaconState state, List<PendingAttestation> attestations) throws IllegalArgumentException {
-
-    List<ArrayList<Integer>> validator_index_sets = new ArrayList<>();
-
-    for (PendingAttestation attestation : attestations) {
-      validator_index_sets.add(
-          get_attestation_participants(
-              state, attestation.getData(), attestation.getAggregation_bitfield().toArray()));
+    HashSet<Integer> output = new HashSet<>();
+    for (PendingAttestation a : attestations) {
+      output.addAll(get_attestation_participants(state, a.getData(), a.getAggregation_bitfield().toArray()));
     }
-
-    List<Integer> attester_indices = new ArrayList<Integer>();
-    for (List<Integer> validator_index_set : validator_index_sets) {
-      for (Integer validator_index : validator_index_set) {
-        if (!attester_indices.contains(validator_index)) {
-          attester_indices.add(validator_index);
-        }
-      }
-    }
-    return attester_indices;
+    List<Integer> output_list = new ArrayList<>(output);
+    Collections.sort(output_list);
+    return output_list;
   }
 
+  /**
+   * @param state
+   * @param attestations
+   * @return
+   */
   public static UnsignedLong get_attesting_balance(
       BeaconState state, List<PendingAttestation> attestations) {
     return get_total_balance(state, get_attesting_indices(state, attestations));
   }
 
+  /**
+   * @param state
+   * @return
+   */
   public static List<PendingAttestation> get_current_epoch_boundary_attestations(
       BeaconState state) {
     List<PendingAttestation> attestations = new ArrayList<>();
-    for (PendingAttestation attestation : state.getCurrent_epoch_attestations()) {
-      if (attestation.getData().getTarget_root()
+    for (PendingAttestation a : state.getCurrent_epoch_attestations()) {
```

Comment:
I'm not a huge fan of one letter variable names outside of simple i, j type iterators in for loops. I think either `attestation` or `pendingAttestation` is more descriptive than `a`.

Reply:
I kept it to match the repo and make it more readable, but happy to switch back.

Reply:
Do you have a strong preference? If so, I'll switch it.